concord-server/server/src/main/java/com/rabidgremlin/concord/ConcordServerApplication.java

```diff
@@ -146,55 +153,66 @@ public void run(ConcordServerConfiguration configuration,
   {
     configureCors(environment);
     environment.jersey().setUrlPattern("/api/*");
-    
-    environment.jersey().register(CsvMessageBodyProvider.class); 
-    
-    
+
+    environment.jersey().register(CsvMessageBodyProvider.class);
+
+
 	// TODO: Clean this up with type system and wrap exceptions
 	Class credentialsValidatorClass = Class.forName(configuration.getCredentialsValidator().getClassName());
 	Constructor credentialsValidatorConstructor = credentialsValidatorClass.getConstructor(HashMap.class);
 	CredentialsValidator credentialsValidator = (CredentialsValidator)credentialsValidatorConstructor.newInstance(configuration.getCredentialsValidator().getConfigProperties());
 
-    
-    environment.jersey().register(new SessionsResource(configuration.getJwtTokenSecret(),credentialsValidator));    
+
+    environment.jersey().register(new SessionsResource(configuration.getJwtTokenSecret(),credentialsValidator));
     environment.jersey().register(new RedirectResource());
-    
+
     final JdbiFactory factory = new JdbiFactory();
     final Jdbi jdbi = factory.build(environment, configuration.getDatabase(), "mysql");
-    
-    
+
+
     // TODO: Very ugly needs to be refactored out
     SystemLabelStore systemLabelStore = new SystemLabelStore() {
-		
+
 		@Override
 		public List<SystemLabel> getSystemLabels() {
 			LabelsDao dao = jdbi.onDemand(LabelsDao.class);
-			
+
 			List<Label> labels = dao.getLabels();
 			ArrayList<SystemLabel> systemLabels = new ArrayList<SystemLabel>();
-			
+
 			for (Label label: labels)
 			{
 				systemLabels.add(new SystemLabel(label.getLabel(), label.getShortDescription(), label.getLongDescription()));
 			}
-			
+
 			return systemLabels;
 		}
 	};
-    
-       
-    
-	// TODO: Clean this up with type system and wrap exceptions
-	Class labelSuggesterClass = Class.forName(configuration.getLabelSuggester().getClassName());
-	Constructor labelSuggesterConstructor = labelSuggesterClass.getConstructor(SystemLabelStore.class);
-	LabelSuggester labelsSuggester = (LabelSuggester)labelSuggesterConstructor.newInstance(systemLabelStore);
-	
-	
-    
+
+    LabelSuggester labelsSuggester = null;
+
+    try
+    {
+      Class labelSuggesterClass = Class.forName(configuration.getLabelSuggester().getClassName());
+      Constructor labelSuggesterConstructor = labelSuggesterClass.getConstructor(SystemLabelStore.class, HashMap.class);
+      labelsSuggester = (LabelSuggester) labelSuggesterConstructor.newInstance(systemLabelStore, configuration.getLabelSuggester().getConfigProperties());
+    }
+    catch (InvocationTargetException e)
+    {
+      if (e.getCause() instanceof InvalidConfigPropertiesException)
+      {
+        log.error("Invalid configuration provided for label suggestor", e);
+      }
+      else
+      {
+        log.error("Unknown error ");

```

Comment:
log error anyway?

Reply:
Yeah I wanted a more specific log message...but may as well just have one and print the exception stacktrace